CONTRIBUTING.md

```diff
@@ -1,55 +1,156 @@
+- In structuring, the documentation employs the [Diátaxis] approach
+
+### Fork and download the documentation repository
+
+If you are working with this documentation set for the first time, you'll need
+to create a fork of this repository on your GitHub account, and clone the
+forked repository.
+
+### Install dependencies
+
+Go into the `docs/` directory and run:
+
+```
 make install
 ```
 
-This will create a virtual environment with all the tooling dedicated to validate and render the documentation.
-It will also install the [woke snap].
+This will create a virtual environment with all the tooling and packages
+required to validate and render the documentation.
 
-## Build and serve the docs
+### Build and serve the docs
 
 To get a live preview and validation of the documentation, run:
 
-```shell
+```
 make run
 ```
 
-This will provide a locally rendered version of the documentation at: http://localhost:8000
+This will provide a locally rendered version of the documentation at:
+<http://localhost:8000>
 
-The rendered preview will get updated every time a document is modified.
+The rendered preview will get updated every time a document is modified and
+saved.
 
-## Validate the docs
+### Validate the docs
 
-You can validate your changes with the following recipes:
+Before you submit a PR, make sure to validate your changes with the following
+recipes:
 
-```shell
+```
 make spelling  
-make linkcheck 
-make woke      # to check for non-inclusive language
+make linkcheck  
+make woke       # to check for non-inclusive language
 ```
 
 ## Propose changes
 
 Submit your documentation changes via a pull request on GitHub.
 
+Your changes will be reviewed in due time; if approved, they will eventually be
+merged.
+
+### Describing pull requests
+
+<!-- TODO: Update with your own checklist or drop if excessive -->
+
+To be properly considered, reviewed, and merged, your pull request must provide
+the following details:
+
+- **Title**: Summarise the change in a short, descriptive title.
+- **Description**: Explain the problem that your pull request solves. Mention
+any new features, bug fixes, or refactoring.
+- **Relevant issues**: Reference any [related issues, pull requests, and
+repositories].
+
+### Commit structure and messages
+
+Use separate commits for each logical change, and for changes to different
+sections in the Real-time Ubuntu documentation.
+Prefix your commit messages with the names of sections or pages that they
+affect, using the code tree structure. For example, start a commit that updates
+the reference page kernel boot parameters with `ref/kernel-boot:`.
+
+Use [conventional commits] to ensure consistency across the project:
+
+```none
+docs(ref/kernel-boot): Add example for nohz parameter
+
+* Additional description goes here
+```
+
+Such structure makes it easier to review contributions and simplifies porting
+fixes to other branches.
+
+### Sign off on commits
```

Comment:
This isn't implemented yet. We can implement this via a GHA workflow check or DCO app.